types/content-type-parser.d.ts

```diff
@@ -0,0 +1,63 @@
+import { Buffer } from "buffer";
+import { RawServerBase, RawServerDefault, RawRequestBase, RawRequestDefault } from './utils';
+
+/**
+ * Body parser method that operatoes on request body
+ */
+export type FastifyBodyParser<
+  RawBody extends string | Buffer,
+  RawServer extends RawServerBase = RawServerDefault,
+  RawRequest extends RawRequestBase = RawRequestDefault<RawServer>,
+> = ((req: RawRequest, rawBody: RawBody, done: (err: Error | null, body?: any) => void) => void)
+  | ((req: RawRequest, rawBody: RawBody) => Promise<any>)
+
+/**
+ * Content Type Parser method that operates on request content
+ */
+export type FastifyContentTypeParser<
+  RawServer extends RawServerBase = RawServerDefault,
+  RawRequest extends RawRequestBase = RawRequestDefault<RawServer>
+> = ((req: RawRequest, done: (err: Error | null, body?: any) => void) => void)
+  | ((req: RawRequest) => Promise<any>)
+
+/**
+ * Natively, Fastify only supports 'application/json' and 'text/plain' content types. The default charset is utf-8. If you need to support different content types, you can use the addContentTypeParser API. The default JSON and/or plain text parser can be changed.
+ */
+export interface addContentTypeParser<
+  RawServer extends RawServerBase = RawServerDefault,
+  RawRequest extends RawRequestBase = RawRequestDefault<RawServer>
+> {
+  (
+    contentType: string | string[],
+    opts: {
+      bodyLimi?: number
```

Comment:
Typo here : replace by `bodyLimit`

Reply:
You should have a test for this interface.

Reply:
contentTypeParser is being finicky now. I may have to issue this out for more investigation.

Reply:
https://stackoverflow.com/questions/56861255/overloaded-typescript-function-parameter-does-not-work-with-async-keyword

Reply:
Resolved in latest commit